src/main/java/uk/gov/hmcts/cmc/claimstore/controllers/support/SupportController.java

```diff
@@ -156,4 +174,32 @@ private void resendStaffNotificationOnAgreementCountersigned(Claim claim) {
         agreementCountersignedStaffNotificationHandler.onAgreementCountersigned(event);
     }
 
+    private void resendClaimsToRPA(List<Claim> claims, String authorisation) {
+        if (StringUtils.isBlank(authorisation)) {
+            throw new BadRequestException("Authorisation is required");
+        }
+
+        for (Claim claim: claims) {
+            GeneratePinResponse pinResponse = userService
+                .generatePin(claim.getClaimData().getDefendant().getName(), authorisation);
+
+            String fullName = userService.getUserDetails(authorisation).getFullName();
+
+            documentGenerator.generateForNonRepresentedRPA(
+                new CitizenClaimIssuedEvent(claim, pinResponse.getPin(), fullName, authorisation)
+            );
+        }
+    }
+
+    private List<Claim> checkClaimsExist(List<String> referenceNumbers) {
+        List<Claim> claims  = new ArrayList<>();
+        for (String referenceNumber: referenceNumbers) {
+            Claim claim = claimService.getClaimByReferenceAnonymous(referenceNumber)
+                .orElseThrow(() -> new NotFoundException(CLAIM + referenceNumber + " does not exist"));
```

Comment:
Not sure what is the requirement here, but this will throw exception if any one of the reference number is incorrect. I hope that is intentional?

Reply:
yes, I don't want it to complete if the set being sent is incorrect

Reply:
👍